### PR TITLE
Fig test slice

### LINKED ISSUE #8: Test with 2D and 3D data
The widgets that take images should work with 2D and 3D data. This should be tested.

2D data:
- [X] Histogram
- [x] Slice
- [x] Scatter

3D data:
- [x] Histogram
- [X] Slice
- [x] Scatter

### PR DESCRIPTION
**Pull request by ruaridhg**
Towards #8 

- Added baseline folder with test images to gitignore
- test_slice used 3D brain data from ski image
- needed to change backend_qt5agg to backend_qtagg (not sure if this is something we want in the base.py since the change was just for testing)

### CODE REVIEW

File: src/napari_matplotlib/tests/test_slice.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+import pytest
 
 from napari_matplotlib import SliceWidget
 
 
-def test_scatter(make_napari_viewer):
-    # Smoke test adding a histogram widget
+@pytest.mark.mpl_image_compare
+def test_slice(make_napari_viewer, brain_data):
+    # Smoke test adding a slice widget
```
Comment (dstansby) [inline src/napari_matplotlib/tests/test_slice.py:8]:
```suggestion
```
This is no longer a smoke test 🎉

File: src/napari_matplotlib/tests/test_slice.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
     viewer = make_napari_viewer()
-    viewer.add_image(np.random.random((100, 100, 100)))
-    SliceWidget(viewer)
+    viewer.add_image(
+        brain_data[0], **brain_data[1]
+    )  # (np.random.random((100, 100, 100)))
```
Comment (dstansby) [inline src/napari_matplotlib/tests/test_slice.py:12]:
```suggestion
    ) 
```
No need to keep old comments around

**Review by dstansby — CHANGES_REQUESTED** [review]
The test is now skipped during testing, and it shouldn't be skipped. I suspect this is because there aren't any baseline images? Instead of gitignoring the test image folder, I think we decided to upload the test images to the repo, so the gitignore needs changing, and the baseline test images adding.

**Comment by ruaridhg** [comment]
> The test is now skipped during testing, and it shouldn't be skipped. (e.g. see https://github.com/matplotlib/napari-matplotlib/actions/runs/4981454254/jobs/8915726820?pr=113#step:7:42)
> 
> I suspect this is because there aren't any baseline images? Instead of gitignoring the test image folder, I think we decided to upload the test images to the repo, so the gitignore needs changing, and the baseline test images adding.

Locally it seems to skip the first time when creating the baseline image and then it works as expected the 2nd time, but clearly is not the case for tox so will investigate further. Yes, I remember now and I've found a 3D image that works then makes sense to remove from gitignore.

**Comment by dstansby** [comment]
👍  - I think the two things that need doing then are:
- Commit the test images to the git repository
- Make sure the test images are available to tox. It might be you need to copy them from the root of the repostiory to the tox environment with a command in the tox file that looks like:
`cp {toxinidir}/path/to/figures/in/repo {envdir}/path/to/figures/in/repo`
(I got the tox variables in curly brackets from https://tox.wiki/en/latest/config.html)

**Review by dstansby — CHANGES_REQUESTED** [review]
The test is still being skipped, I'm guessing because it still isn't finding the test image?

**Comment by ruaridhg** [comment]
> The test is still being skipped, I'm guessing because it still isn't finding the test image?

Yes, I've not got it quite working yet as using tox locally, the test_slice is being skipped due to `Unable to download data/brain.tiff` so I need to locate that and copy it over to tox

**Comment by ruaridhg** [comment]
@dstansby Turns out `python -c 'from skimage import data; data.brain()'` was the command needed to make the data available to tox. Should be fine to merge now, however Read the Docs build failed

File: .gitignore [hunk 3]
```diff
@@ -37,6 +37,7 @@ pip-delete-this-directory.txt
 # Unit test / coverage reports
 htmlcov/
 .tox/
+report/
```
Comment (dstansby) [inline .gitignore:40]: Out of interest, what's this for?

File: tox.ini [hunk 4]
```diff
@@ -11,4 +11,6 @@ python =
 
 [testenv]
 extras = testing
-commands = python -m pytest --mpl -v --color=yes --cov=napari_matplotlib --cov-report=xml
+commands =
+    - python -c 'from skimage import data; data.brain()'
```
Comment (dstansby) [inline tox.ini:15]: Given this import and call is in `conftest.py`, I'm not sure why it needs to be here too. Do the tests fail without this line in `tox.ini`?

File: .gitignore (hunk 3 above)
Comment (ruaridhg) [reply .gitignore:40]: It's the pytest-mpl report images which I have locally, but should include as a build artefact

**Review by dstansby — APPROVED** [review]

**Pull request merged**